### PR TITLE
Fix LinearVisitor behavior for comparison nodes, add tests

### PR DESCRIPTION
**Pull request by flomnes**

### CODE REVIEW

File: src/tests/src/solver/expressions/test_expressions.cpp [hunk 1]
```diff
@@ -428,14 +428,68 @@ BOOST_FIXTURE_TEST_CASE(comparison_nodes_are_not_linear, Registry<Node>)
...
+
+BOOST_FIXTURE_TEST_CASE(comparison_nodes_variable_constant_is_linear, Registry<Node>)
+{
+    PrintVisitor printVisitor;
+    LinearityVisitor linearVisitor;
+
+    VariableNode var1("x");
+    // variable
+    LiteralNode literal(21.);
+    // x==y
```
Comment (guilpier-code) [inline src/tests/src/solver/expressions/test_expressions.cpp]:
```C++
// x==y
```
?
Do we mean : 
```C++
// x=21
```
instead ?
(I didn't know we could meet a '=='  in an expression, could we ?)

File: src/tests/src/solver/expressions/test_expressions.cpp [hunk 2]
```diff
@@ -428,14 +428,68 @@ BOOST_FIXTURE_TEST_CASE(comparison_nodes_are_not_linear, Registry<Node>)
...
     BOOST_CHECK_EQUAL(printVisitor.dispatch(*lt), "x<=y");
-    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*lt), LinearStatus::NON_LINEAR);
+    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*lt), LinearStatus::LINEAR);
     // x>=y
     Node* gt = create<GreaterThanOrEqualNode>(&var1, &var2);
     BOOST_CHECK_EQUAL(printVisitor.dispatch(*gt), "x>=y");
+    BOOST_CHECK_EQUAL(linearVisitor.dispatch(*gt), LinearStatus::LINEAR);
+}
+
+BOOST_FIXTURE_TEST_CASE(comparison_nodes_variable_constant_is_linear, Registry<Node>)
```
Comment (guilpier-code) [inline src/tests/src/solver/expressions/test_expressions.cpp]: Test name **comparison_nodes_variable_constant_is_linear** : 
Do we want a comparison ? Or an assignment ?

**Review by guilpier-code — CHANGES_REQUESTED** [review]

**Comment by a-zakir** [comment]
what a shame (me) , <=, >= are not for comparison purpose but as 'sign', to represent constraints, variables bounds ...

**Comment by flomnes** [comment]
> what a shame (me) , <=, >= are not for comparison purpose but as 'sign', to represent constraints, variables bounds ...

It's OK, the spec wasn't so clear

File: src/tests/src/solver/expressions/test_expressions.cpp (hunk 1 above)
Comment (flomnes) [reply src/tests/src/solver/expressions/test_expressions.cpp]: yes

File: src/tests/src/solver/expressions/test_expressions.cpp (hunk 2 above)
Comment (flomnes) [reply src/tests/src/solver/expressions/test_expressions.cpp]: These are all comparisons

**Review by guilpier-code — APPROVED** [review]

**Pull request merged**